### PR TITLE
Prover: bump go-corset to the latest version

### PR DESCRIPTION
**Pull request by gusiri**
This PR
- bumps go-corset to the latest commit as this fixes some performance issues exposed by this bug 82f81b5.
- fixes compile error with the latest go-corset (String() undefined)
```
zkevm/arithmetization/definition.go:90:20: corsetCS.String undefined (type "github.com/consensys/go-corset/pkg/schema".Constraint has no field or method String)
zkevm/arithmetization/definition.go:234:57: e.String undefined (type air.Expr has no field or method String)
```
- fixes golangci-lint version to v1.61.0
- fixes error code 77 message format (https://github.com/Consensys/linea-monorepo/issues/162)

### Checklist

* [ ] I wrote new tests for my new core changes.
* [x] I have successfully ran tests, style checker and build against my new changes locally.
* [x] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

**Comment by gusiri** [comment]
I am looking at some errors:
```
run golangci-lint
  Running [/home/runner/golangci-lint-1.61.0-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=prover --timeout=5m] in [/runner/_work/linea-monorepo/linea-monorepo/prover] ...
  Error: could not import github.com/consensys/linea-monorepo/prover/zkevm/arithmetization (-: # github.com/consensys/linea-monorepo/prover/zkevm/arithmetization
  Error: zkevm/arithmetization/definition.go:90:20: corsetCS.String undefined (type "github.com/consensys/go-corset/pkg/schema".Constraint has no field or method String)
  Error: zkevm/arithmetization/definition.go:234:57: e.String undefined (type air.Expr has no field or method String)) (typecheck)
  Error: could not import github.com/consensys/linea-monorepo/prover/zkevm/arithmetization (-: # github.com/consensys/linea-monorepo/prover/zkevm/arithmetization
  Error: zkevm/arithmetization/definition.go:90:20: corsetCS.String undefined (type "github.com/consensys/go-corset/pkg/schema".Constraint has no field or method String)
  Error: zkevm/arithmetization/definition.go:234:57: e.String undefined (type air.Expr has no field or method String)) (typecheck)
  Error: : # github.com/consensys/linea-monorepo/prover/zkevm/arithmetization [github.com/consensys/linea-monorepo/prover/zkevm/arithmetization.test]
  Error: zkevm/arithmetization/definition.go:90:20: corsetCS.String undefined (type "github.com/consensys/go-corset/pkg/schema".Constraint has no field or method String)
  Error: zkevm/arithmetization/definition.go:234:57: e.String undefined (type air.Expr has no field or method String) (typecheck)
```

**Comment by AlexandreBelling** [comment]
Is it the linter or also the whole compilation?

**Review by Soleimani193 — APPROVED** [review]

File: prover/zkevm/arithmetization/definition.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func (s *schemaScanner) scanConstraints() {
 	corsetCSs := s.Schema.Constraints().Collect()
 
 	for _, corsetCS := range corsetCSs {
-		name := corsetCS.String()
+		name := fmt.Sprintf("%v", corsetCS)
```
Comment (gusiri) [inline prover/zkevm/arithmetization/definition.go:90]: @AlexandreBelling 
There were actually compile errors with the latest go-corset.  
```
zkevm/arithmetization/definition.go:90:20: corsetCS.String undefined (type "github.com/consensys/go-corset/pkg/schema".Constraint has no field or method String)
```

File: prover/zkevm/arithmetization/definition.go [hunk 2]
```diff
@@ -230,8 +230,8 @@ func (s *schemaScanner) castExpression(expr air.Expr) *symbolic.Expression {
 		return symbolic.NewVariable(c)
 
 	default:
-
-		panic(fmt.Sprintf("unsupported type: %T for %v", e, e.String()))
+		eStr := fmt.Sprintf("%v", e)
```
Comment (gusiri) [inline prover/zkevm/arithmetization/definition.go:233]: Here as well.
```
zkevm/arithmetization/definition.go:234:57: e.String undefined (type air.Expr has no field or method String)
```

File: .github/workflows/prover-testing.yml [hunk 3]
```diff
@@ -44,6 +44,7 @@ jobs:
     - name: golangci-lint
       uses: golangci/golangci-lint-action@v3
       with:
+          version: v1.61.0
```
Comment (gusiri) [inline .github/workflows/prover-testing.yml:47]: Also, I fixed `golangci-lint` version to `v1.61.0` (latest)

**Review by AlexandreBelling — APPROVED** [review]

**Pull request merged**